src/main/java/com/khutircraftubackend/services/UserService.java

```diff
@@ -0,0 +1,168 @@
+package com.khutircraftubackend.services;
+
+import com.khutircraftubackend.dto.UserDTO;
+import com.khutircraftubackend.dto.security.PasswordUpdateRequest;
+import com.khutircraftubackend.mapper.UserMapper;
+import com.khutircraftubackend.models.Role;
+import com.khutircraftubackend.models.User;
+import com.khutircraftubackend.repositories.UserRepository;
+import com.khutircraftubackend.security.JwtUtils;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.mail.SimpleMailMessage;
+import org.springframework.mail.javamail.JavaMailSender;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.stereotype.Service;
+
+import java.util.Optional;
+import java.util.UUID;
+
+/**
+ * Клас UserService реалізує бізнес-логіку для роботи з користувачами.
+ * <p>
+ * Цей клас містить методи для створення, оновлення, видалення та отримання користувачів.
+ * </p>
+ */
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class UserService {
+
+    private final UserRepository userRepository;
+    private final PasswordEncoder passwordEncoder;
+    private final JwtUtils jwtUtils;
+    private final JavaMailSender mailSender;
+    private final SendinblueEmailService sendinblueEmailService;
+
+    /**
+     * Реєструє нового користувача.
+     * <p>
+     * Метод перетворює UserDTO в User, хешує пароль, генерує JWT токен та код підтвердження.
+     * Потім зберігає користувача в базі даних і надсилає електронного листа з кодом підтвердження.
+     * </p>
+     *
+     * @param userDTO дані для реєстрації користувача
+     * @return UserDTO об'єкт зареєстрованого користувача
+     */
+
+    @Transactional
+    public UserDTO registerNewUser(UserDTO userDTO) {
+        User user = UserMapper.INSTANCE.userDTOToUser(userDTO);
+
+        if(userDTO.isPasswordMatching()) {
+            user.setPassword(passwordEncoder.encode(user.getPassword()));
+            user.setRole(userDTO.role() == Role.SELLER ? Role.SELLER : Role.BUYER);
```

Comment:
i'd suggest compare these with Object equality approach, using `whatever.equal()` method.

Reply:
I do not understand. If we are talking about roles, maybe switch is better?